ql/methods/finitedifferences/tridiagonaloperator.hpp

```diff
@@ -178,14 +178,13 @@ namespace QuantLib {
             timeSetter_->setTime(t, *this);
     }
 
-    inline void TridiagonalOperator::swap(TridiagonalOperator& from) {
-        using std::swap;
-        swap(n_, from.n_);
+    inline void TridiagonalOperator::swap(TridiagonalOperator& from) noexcept {
+        std::swap(n_, from.n_);
         diagonal_.swap(from.diagonal_);
         lowerDiagonal_.swap(from.lowerDiagonal_);
         upperDiagonal_.swap(from.upperDiagonal_);
         temp_.swap(from.temp_);
-        swap(timeSetter_, from.timeSetter_);
+        std::swap(timeSetter_, from.timeSetter_);
```

Comment:
Right, I should have recognized that, sorry. But I generally find it more clear to avoid this idiom wherever possible. Looking at the [boost docs](https://www.boost.org/doc/libs/1_82_0/libs/smart_ptr/doc/html/smart_ptr.html#shared_ptr_free_functions), the free swap function for `boost::shared_ptr` just calls `a.swap(b)`, so perhaps it would be better to just call `timeSetter_.swap(from.timeSetter_)` here. Or I can add back the idiom everywhere if you prefer, just let me know.

Reply:
I guess `timeSetter_.swap(from.timeSetter_)` is fine&mdash;this way we don't even have to rely on the compiler inlining the free swap function.

Reply:
Ok, I have changed to `timeSetter_.swap(from.timeSetter_)` and have also changed two other places in `NinePointLinearOp::swap` and `TripleBandLinearOp::swap` where `std::swap` was used instead of the swap member function.